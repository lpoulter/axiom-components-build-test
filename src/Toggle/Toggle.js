import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './Toggle.css';

export default class Toggle extends Component {
  static propTypes = {
    /** Set alignment of the switch **/
    align: PropTypes.oneOf(['left', 'right', 'full']),
    /** Label that is inserted next to the toggle switch */
    children: PropTypes.node,
    /** Disabled control of the switch */
    disabled: PropTypes.bool,
    /** Called when the toggle is toggled/changed */
    onToggle: PropTypes.func.isRequired,
    /** Size of the toggle */
    size: PropTypes.oneOf(['small', 'medium']),
    /** Toggled/checked state of the switch */
    toggled: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    align: 'left',
    size: 'small',
  };

  render() {
    const { align, children, disabled, size, toggled, onToggle, ...rest } = this.props;
    const classes = classnames('ax-toggle', `ax-toggle--${size}`, `ax-toggle--align-${align}`, {
      'ax-toggle--enabled': !disabled,
    });

    return (
      <Base Component="label" className={ classes } space="x2">
        { children && (
          <span className="ax-toggle__label">
            { children }
          </span>
        ) }

        <span className="ax-toggle__switch">
          <input { ...rest }
              checked={ toggled }
              className="ax-toggle__input"
              disabled={ disabled }
              onChange={ onToggle }
              type="checkbox" />
          <span className="ax-toggle__appearance" />
        </span>
      </Base>
    );
  }
}
