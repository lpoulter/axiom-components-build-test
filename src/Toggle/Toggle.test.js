import React from 'react';
import renderer from 'react-test-renderer';
import Toggle from './Toggle';

const onToggle = () => {};

describe('Toggle', () => {
  it('renders as toggled', () => {
    const tree = renderer.create(<Toggle onToggle={ onToggle } toggled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders as not toggled', () => {
    const tree = renderer.create(<Toggle onToggle={ onToggle } toggled={ false } />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with disabled', () => {
    const tree = renderer.create(
      <Toggle
          disabled
          onToggle={ onToggle }
          toggled={ false } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with label children', () => {
    const tree = renderer.create(
      <Toggle onToggle={ onToggle } toggled={ false }>Toggle</Toggle>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size small', () => {
    expect(renderer.create(
      <Toggle onToggle={ onToggle } size="small" toggled={ false } />
    ).toJSON()).toMatchSnapshot();
  });

  it('renders with size medium', () => {
    expect(renderer.create(
      <Toggle onToggle={ onToggle } size="medium" toggled={ false } />
    ).toJSON()).toMatchSnapshot();
  });

  it('renders with align left', () => {
    expect(renderer.create(
      <Toggle align="left" onToggle={ onToggle } toggled={ false } />
    ).toJSON()).toMatchSnapshot();
  });

  it('renders with align right', () => {
    expect(renderer.create(
      <Toggle align="right" onToggle={ onToggle } toggled={ false } />
    ).toJSON()).toMatchSnapshot();
  });

  it('renders with align full', () => {
    expect(renderer.create(
      <Toggle align="full" onToggle={ onToggle } toggled={ false } />
    ).toJSON()).toMatchSnapshot();
  });
});
