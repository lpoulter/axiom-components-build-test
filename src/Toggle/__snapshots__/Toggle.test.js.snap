// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders as not toggled 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders as toggled 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={true}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with align full 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-full ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with align left 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with align right 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-right ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with disabled 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with label children 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__label"
  >
    Toggle
  </span>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with size medium 1`] = `
<label
  className="ax-toggle ax-toggle--medium ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;

exports[`Toggle renders with size small 1`] = `
<label
  className="ax-toggle ax-toggle--small ax-toggle--align-left ax-toggle--enabled ax-space--x2"
>
  <span
    className="ax-toggle__switch"
  >
    <input
      checked={false}
      className="ax-toggle__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ax-toggle__appearance"
    />
  </span>
</label>
`;
